File: java/src/main/java/Pages/MainPage.java
package WebDriverPackage;

import Utils.TestLogger;
import org.apache.log4j.Logger;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.io.IOException;
import java.net.MalformedURLException;

public class MainPage {

    private String url= "http://przyklady.javastart.pl/jpetstore/";
    private WebDriverWrapper webDriverWrapper;
    private TestLogger testLogger;

    private static final String enterStoreLink = "Enter the Store";
    @FindBy(linkText = enterStoreLink)
    private WebElement enterStore;

    private static final String searchButtonName = "searchProducts";
    @FindBy(name = searchButtonName)
    private WebElement searchButton;

    private static final String searchFiledName = "keyword";
    @FindBy(name = searchFiledName)
    private WebElement searchFiled;

    private static final String table = "table";
    @FindBy(css =table)
    private WebElement productTable;

    public TestLogger getLogger() {
       return testLogger;
    }
    public MainPage(TestLogger testLogger) {
        try {
            webDriverWrapper = new WebDriverWrapper(testLogger,false);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
    }
    public MainPage openMainPage() {
        PageFactory.initElements(webDriverWrapper.getDriver(), this);
        webDriverWrapper.OpenPage(url);
        return this;
    }
    public WebDriverWrapper getWebDriverWrapper() {
        return webDriverWrapper;
    }
    public void ClickEnterStoreLink() {
        webDriverWrapper.clickOnElement(enterStore);
    }
    public void ClickSearchButton() {
        webDriverWrapper.clickOnElement(searchButton);
    }
    public void enterSearchText(String productName){
        webDriverWrapper.enterText(searchFiled, productName);
    }
    public boolean checkIfProductTableIsDisplay() {

        try {
            webDriverWrapper.waitUntilElementIsVisible(productTable);
            return true;
        }
        catch (Exception ex) {
            testLogger.getLog().error(ex.getMessage());
            return false;
        }
    }
    public boolean waitForProductLink(String productName){
        try {
            webDriverWrapper.takeScreenshot(productName);
            WebElement product = webDriverWrapper.findElementByLinkText(productName);
            webDriverWrapper.waitUntilElementIsVisible(product);
            return true;
        }
        catch (Exception ex) {
            testLogger.getLog().error(ex.getMessage());
            try {
                webDriverWrapper.takeScreenshot("Exeption");
            } catch (IOException e) {
                e.printStackTrace();
            }
            return false;
        }
    }
}

File: python/Shop_Tests/pages/main_page.py
from selenium.webdriver.common.by import By

from Shop_Tests.webdriver_wrapper.webdriver_wrapper import WebDriverWrapper


class MainPage:
    url = "http://przyklady.javastart.pl/jpetstore/"
    enter_store_link = (By.LINK_TEXT, "Enter the Store")
    search_button = (By.NAME, "searchProducts")
    search_field = (By.NAME, "keyword")

    def __init__(self, logging, web_driver_wrapper: WebDriverWrapper):
        self.logging = logging
        self.web_driver_wrapper = web_driver_wrapper

    def open_page(self):
        self.logging.log().info("Open page %s" % MainPage.url)
        self.web_driver_wrapper.open_page(MainPage.url)

    def click_enter_store_link(self):
        self.logging.log().info("Click enter store link")
        self.web_driver_wrapper.driver.find_element(*MainPage.enter_store_link).click()
        return self

    def click_search_button(self):
        self.logging.log().info("Click search button")
        self.web_driver_wrapper.driver.find_element(*MainPage.search_button).click()
        return self

    def enter_search_text(self, text: str):
        self.logging.log().info("Enter text: %s" % text)
        search_field = self.web_driver_wrapper.driver.find_element(
            *MainPage.search_field
        )
        search_field.clear()
        search_field.send_keys(text)

        return self

    def check_if_link_is_displayed(self, link: str):
        try:
            self.web_driver_wrapper.wait_for_element(By.LINK_TEXT, link)
            return True
        except Exception as ex:
            self.logging.error(ex)
            return False
